### DISCUSSION TITLE
[parse_url] unable to parse url: relative URL without a base"

### DISCUSSION THREAD

**Discussion by StefanSa** [vectordotdev/vector#11745] [Q&A]
Hi there,
I get this error message with such URLs `test.com`:
```
Mapping failed with event. error="function call error for \"parse_url\" at (78:130): unable to parse url: relative URL without a base" internal_log_rate_secs=30
WARN transform{component_kind="transform" component_id=ecs_url_mapping_atp component_type=remap component_name=ecs_url_mapping_atp}: vector::internal_events::remap: Internal log [Mapping failed with event.] is being rate limited.
```
How do i handle something like this ?
I would like to be able to parse such URLs.

**Comment by jszwedko** [comment]
Hi @StefanSa !

Do you mean you want to parse that domain name into parts? Or do you mean you want to parse that domain name as a URL?

**Reply by JeanMertz** [reply]
This is an unfortunate limitation of the internal library we use to parse URLs. It always requires "absolute paths", not relative ones, and in this case, it needs a base (e.g. the scheme `https` for example) to know that it is dealing with the domain `test.com`.

I guess we could work around that restriction by checking for this specific error, and manually adding the `https` scheme if needed. We'll probably want to make this optional with some kind of `strict: false` parameter, as it might otherwise confuse others who don't expect this behaviour.
